### DISCUSSION TITLE
With connection port +1, is it possible to connect to a server with steam://connect?

### DISCUSSION THREAD

**Discussion by miend** [mbround18/valheim-docker#405] [Q&A]
I'm trying to do a self-host using this container setup, and I can connect in-game at `[IP]:[port]` just fine (where `PORT` is `2456`).

What I'd like to be able to do is connect with the steam procotol at `steam://connect/[ip]:[port]/[password]`. Attempting to connect with port 2456 results in a failure to find the game server, which makes sense. Attempting to connect with 2457 results in all the right Steam metadata being returned, the game launching, character selection, however it then fails to connect to server as, I imagine, it's trying to connect at port 2457. Is there any conceivable way I might make a `steam://connect` link function for Valheim/this setup?

**Comment by mbround18** [comment]
@miend , yes this should work, its the primary connection method I use for my server instance. Try dropping the password portion.

**Reply by miend** [reply]
@mbround18 I've done so, but no luck with steam://connect. The only way I've been able to connect is via steam://run populated with the connection details. This is unfortunate since it requires users to have the game closed before connecting. What port do you connect on? Is there anything particular to your setup that enables this? When looking for similar issues before, I ran across a couple comments like [this](https://valheimbugs.featureupvote.com/suggestions/160837/valve-steam-connect-not-working) which suggest I'm not the only one with this issue, and I haven't yet found a resolution for it. Would you be willing to share any more potentially relevant details about your setup and how steam://connect performs for you?

Thanks for this Docker build by the way. I've been using it to PoC a web (microservices)-driven setup on GCP to let friends and I play at any time with minimal monthly cost -- hence the long time between replies here.

**Reply by mbround18** [reply]
> 
> 
> @mbround18 I've done so, but no luck with steam://connect. The only way I've been able to connect is via steam://run populated with the connection details. This is unfortunate since it requires users to have the game closed before connecting. What port do you connect on? Is there anything particular to your setup that enables this? When looking for similar issues before, I ran across a couple comments like [this](https://valheimbugs.featureupvote.com/suggestions/160837/valve-steam-connect-not-working) which suggest I'm not the only one with this issue, and I haven't yet found a resolution for it. Would you be willing to share any more potentially relevant details about your setup and how steam://connect performs for you?
> 
> Thanks for this Docker build by the way. I've been using it to PoC a web (microservices)-driven setup on GCP to let friends and I play at any time with minimal monthly cost -- hence the long time between replies here.

Hey my sincerest apologies for taking a while to get back to you, were you able to figure out the connection urls?

**Comment by lyk4ndna** [comment]
Try this out to launch Valheim directly from desktop, if you have the Valheim desktop shortcut you can right clip and copy the shortcut and paste a new one. 

Right click that new shortcut and click properties.

It should say this: steam://rungameid/892970

Change it to below with your public IP address. It should launch just fine.

steam://rungameid/892970//+connect [insert IP]:2456

If you don't have the Valheim shortcut, then just create a new short cut with the above URL and it'll launch into Valheim directly, select character, type password for server and you're in.
